STAC-16111: Upgrade KOTS documentation

viliakov opened the pull request:
Updating the KOTS docs: adding the latest configuration settings

gggina commented on setup/install-stackstate/kots-install/cluster_management.md:
````diff
@@ -82,6 +82,26 @@ NODES=$(kubectl get ${NODE_TYPE} -n ${LONGHORN_NAMESPACE} -o name)
 kubectl patch -n ${LONGHORN_NAMESPACE} ${NODES} --type='json'  -p "[{\"op\": \"replace\", \"path\": \"/spec/disks/default-disk-ca1000000000/storageReserved\", \"value\": ${STORAGE_RESERVED}}]"
 ```
+
+### Changing the number of replicas for Longhorn persistent volumes
````
Use "Change" rather than "Changing" - this works better in e.g. search.

viliakov replied:
Fixed

gggina commented on setup/install-stackstate/kots-install/cluster_management.md:
````diff
@@ -82,6 +82,26 @@ NODES=$(kubectl get ${NODE_TYPE} -n ${LONGHORN_NAMESPACE} -o name)
+kubectl delete -n default statefulsets -l kots.io/app-slug=stackstate
+```
+
+2. Select the required storage class in [Persistent volume settings](/setup/install-stackstate/kots-install/kots_configuration_screen.md#persistent-volume-settings) section of the KOTS Config UI.
````
2. Select the required storage class in the [Persistent volume settings]...

gggina commented on setup/install-stackstate/kots-install/cluster_management.md:
```diff
@@ -82,6 +82,26 @@ NODES=$(kubectl get ${NODE_TYPE} -n ${LONGHORN_NAMESPACE} -o name)
+
+2. Select the required storage class in [Persistent volume settings](/setup/install-stackstate/kots-install/kots_configuration_screen.md#persistent-volume-settings) section of the KOTS Config UI.
+
+3. Redeploy the application via KOTS Config UI.
```
3. Redeploy the application via the KOTS Config UI.

gggina commented on setup/install-stackstate/kots-install/cluster_management.md:
```diff
@@ -82,6 +82,26 @@ NODES=$(kubectl get ${NODE_TYPE} -n ${LONGHORN_NAMESPACE} -o name)
+
+3. Redeploy the application via KOTS Config UI.
+
+4. Change the number of replicas for the following persistent volumes on the *Volume* tab of the Longhorn UI:
```
Use bold in place of italic (we aim to never use italic)

`4. Change the number of replicas for the following persistent volumes on the **Volume** tab of the Longhorn UI:`

gggina commented on setup/install-stackstate/kots-install/cluster_management.md:
```diff
@@ -82,6 +82,26 @@ NODES=$(kubectl get ${NODE_TYPE} -n ${LONGHORN_NAMESPACE} -o name)
+
+### Changing the number of replicas for Longhorn persistent volumes
+
+1. Scale down all the Stackstate deployments and delete all statefulsets. The deletion is required since the immutable fields of the statefulsets have to be changed. The data stored on the persistent volumes won't be deleted.
```
Lines that are followed by an  associated command or steps should end with a colon `:`
(check all steps on this page for this). For example:

`1. Scale down all the Stackstate deployments and delete all statefulsets. The deletion is required since the immutable fields of the statefulsets have to be changed. The data stored on the persistent volumes won't be deleted:`

gggina commented on setup/install-stackstate/kots-install/cluster_management.md:
```diff
@@ -82,6 +82,26 @@ NODES=$(kubectl get ${NODE_TYPE} -n ${LONGHORN_NAMESPACE} -o name)
+3. Redeploy the application via KOTS Config UI.
+
+4. Change the number of replicas for the following persistent volumes on the *Volume* tab of the Longhorn UI:
+1 - for the `longhorn-single-replica` storage class.
```
if you indent the next line and use a * you will get a nice bulleted sub-list, which will be easier to follow:

```
4. Change the number of replicas for the following persistent volumes on the **Volume** tab of the Longhorn UI:
   * 1 - for the `longhorn-single-replica` storage class.
   * 2 - for the `longhorn-stackstate` (two-replicas volumes)
 ```

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -15,18 +15,25 @@ Kubernetes Off The Shelf (KOTS) allows embedded Kubernetes to run clusters on VM
 
-StackState can be accessed via HTTP and/or HTTPS.  The hostname should be the IP or URL to the master server without `http://` or `https://`.
+StackState can be accessed via HTTP and/or HTTPS. The hostname should be the IP or URL to the master server without `http://` or `https://`.
+If needed HTTP proxy can be used for event handlers. The HTTP proxy hostname should be either FQDN or IP-address.
```
If needed, HTTP proxy can ...

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -15,18 +15,25 @@ Kubernetes Off The Shelf (KOTS) allows embedded Kubernetes to run clusters on VM
+
+### Persistent volume settings
+
+By default the Stackstate application uses a single-replica persistent volumes backed up by [Longhorn](https://longhorn.io/docs/). The data redundancy for StackState components like Kafka, Zookeper, HDFS, and Elasticsearch can be achieved by switching to two-replicas volumes.
```
By default, the Stackstate application ...

gggina replied:
When we link to an external site, we always name the URL we are linking to (so it's clear that clicking the link will take you away from our site):

```
persistent volumes backed up by [Longhorn \(longhorn.io\)](https://longhorn.io/docs/).
```

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
 
+### Authentication
+
+Stackstate supports different authentication backends: File-based (default), LDAP, OIDC, Keycloak. For more information on how to configure them please refer to [Authentication options](https://docs.stackstate.com/configure/security/)
```
In a list the last item needs "and":

`Stackstate supports different authentication backends: File-based (default), LDAP, OIDC and Keycloak.`

gggina replied:
Don't use "please":

`For more information on how to configure them, refer to [Authentication options](https://docs.stackstate.com/configure/security/)`

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+
+### Automated backup
+
+This section allows to setup automatic Cronjob backup for Elasticsearch and StackGraph.
```
Change to this:
`This section explains how to set up an automatic Cronjob backup for Elasticsearch and StackGraph.`

Also, I checked - it seems Cronjob really should begin with a capital C :)

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+### Automated backup
+
+This section allows to setup automatic Cronjob backup for Elasticsearch and StackGraph.
+A number of storage options for the backups is available: AWS S3, Azure Blob Storage, and NFS.
```
Change to this:
`A number of storage options are available for the backups: AWS S3, Azure Blob Storage, and NFS.`

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+This section allows to setup automatic Cronjob backup for Elasticsearch and StackGraph.
+A number of storage options for the backups is available: AWS S3, Azure Blob Storage, and NFS.
+
+The backup needs a Kubernetes persistent volume, which might be configured as a single- or two-replicas Longhorn volume. Keep in mind, that the switching between the replicas mode is not possible without the backup persistent volume removal.
```
Change to this:
```
The backup needs a Kubernetes persistent volume. This can be configured as a single- or two-replicas Longhorn volume. Keep in mind that it is not possible to switch between the replicas mode without removing the backup persistent volume.
```

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+The backup needs a Kubernetes persistent volume, which might be configured as a single- or two-replicas Longhorn volume. Keep in mind, that the switching between the replicas mode is not possible without the backup persistent volume removal.
+
+#### AWS
+AWS backup requires AWS user credentials (AWS access and secret keys) with the read/write permissions on S3 buckets. The backups for Elasticsearch and StackGraph can be stored to different S3 buckets.
```
`can be stored in`...

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+![](/.gitbook/assets/kots-backup-aws.png)
+
+#### Azure Blob Storage
+Azure backup requires setting Azure storage account name, storage account key, and Blob containers for both Elasticsearch and StackGraph backups
```
Use this:
`Azure backup requires setting the Azure storage account name, storage account key and Blob containers for both Elasticsearch and StackGraph backups.`

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+![](/.gitbook/assets/kots-backup-azure.png)
+
+#### NFS
+NFS backup requires an NFS server's FQDN or IP-address, the NFS path to mount, and paths within the server for Elasticsearch and StackGraph backups. If needed low-level NFS parameters like *mount options* and *uid*/*gid* can be set. For this type of backup a Kubernetes persistent volume of type NFS is provisioned.
```
`NFS backup requires the NFS server's FQDN or IP-address, the NFS path to mount and paths within the server for Elasticsearch and StackGraph backups.`

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+
+### SSL Settings
+
+StackState has several points of interaction with external systems, for example event handlers can call out to webhooks in other systems while plugins can retrieve data from external systems like Splunk or Elasticsearch. With the default configuration, StackState will not be able to communicate with these systems when they are secured with TLS using a self-signed certificate or a certificate that ...
```
`StackState has several points of interaction with external systems. For example, event handlers can call out to webhooks in other systems and plugins can retrieve data from external systems such as AWS or Elasticsearch.`

I removed the reference to Splunk plugin as it's just so confusing. The splunk integration doesn't use it.

gggina commented on setup/install-stackstate/kots-install/kots_configuration_screen.md:
```diff
@@ -44,9 +51,45 @@ Once these are all configured, click **Save Config** and follow the prompts to d
+### SSL Settings
+
+StackState has several points of interaction with external systems, for example event handlers can call out to webhooks in other systems while plugins can retrieve data from external systems like Splunk or Elasticsearch. With the default configuration, StackState will not be able to communicate with these systems when they are secured with TLS using a self-signed certificate or a certificate that ...
+To mitigate this, StackState allows configuration of a custom Java trust store. For more information please refer to [Create a custom trust store](https://docs.stackstate.com/configure/security/self-signed-certificates#create-a-custom-trust-store)
```
don't use please

gggina reviewed (commented):
great!

viliakov commented:
Hi @gggina , I addresses all the comments. Found one more "please", and remove it: `Please be aware that changing these settings` -> `Be aware that changing these settings` on 34th line of kots_configuration_screen.md.
Thank you.
